queue_publisher.py

```diff
@@ -65,22 +65,25 @@ def _disconnect(self):
         except Exception as e:
             self.logger.error("Unable to close connection", exception=repr(e))
 
-    def _publish(self, message):
+    def _publish(self, message, tx_id):
```

Comment:
The message _is_ the `tx_id` so no need to pass the same thing twice. Maybe better to just rename the `message` parameter, or pass `message` into the `tx_id` header.